krystex/src/test/java/com/flipkart/krystal/krystex/node/KrystalNodeExecutorTest.java

```diff
@@ -50,6 +52,44 @@ void tearDown() {
     this.krystalNodeExecutor.close();
   }
 
+  /** Executing same node multiple times in a single execution */
+  @Test
+  void multiRequestExecution() throws Exception {
+    NodeDefinition nodeDefinition =
+        nodeDefinitionRegistry.newNodeDefinition(
+            "node",
+            newComputeLogic(
+                    "nodeLogic", Collections.emptySet(), dependencyValues -> "computed_value")
+                .nodeLogicId());
+
+    CompletableFuture<Object> future_1 =
+        krystalNodeExecutor.executeNode(nodeDefinition.nodeId(), Inputs.empty(), "req_1");
+    CompletableFuture<Object> future_2 =
+        krystalNodeExecutor.executeNode(nodeDefinition.nodeId(), Inputs.empty(), "req_2");
+
+    krystalNodeExecutor.flush();
+    assertEquals("computed_value", timedGet(future_1));
+    assertEquals("computed_value", timedGet(future_2));
+  }
+
+  @Test
+  void multiRequestExecutionWithNullRequestId() throws Exception {
+    NodeDefinition nodeDefinition =
+        nodeDefinitionRegistry.newNodeDefinition(
+            "node",
+            newComputeLogic(
+                    "nodeLogic", Collections.emptySet(), dependencyValues -> "computed_value")
+                .nodeLogicId());
+
+    CompletableFuture<Object> future_1 =
+        krystalNodeExecutor.executeNode(nodeDefinition.nodeId(), Inputs.empty(), "req_1");
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> krystalNodeExecutor.executeNode(nodeDefinition.nodeId(), Inputs.empty(), null));
```

Comment:
I think you removed exception message assertion. Was this intentional?
If not, you can try

```java
assertThatThrownBy(() -> executeNode)
        .isInstanceOf(IllegalArgumentException.class)
        .hasMessageContaining()
```